src/http/one/Parser.cc

```diff
@@ -256,6 +256,11 @@ Http::One::Parser::getHeaderField(const char *name)
         // prevent buffer overrun on char header[];
         p.chop(0, sizeof(header)-1);
 
+        // XXX: currently only used for pre-parse Host header, ensure valid domain[:port] or ip[:port]
+        static const CharacterSet hostValid = CharacterSet("host",":[].-_") + CharacterSet::ALPHA + CharacterSet::DIGIT;
```

Comment:
If possible, please use auto.

I also suggest renaming hostValid to hostChars for consistency sake (many such sets are named using fooChars and similar patterns) and to avoid the implication that all those chars are valid in a host name.

```suggestion
        static const auto hostChars = CharacterSet("host",":[].-_") + CharacterSet::ALPHA + CharacterSet::DIGIT;
```

Reply:
Done